### DISCUSSION TITLE
DesktopVK - Suggestions For Collating & Tackling Issues

### DISCUSSION THREAD

**Discussion by squarebananas** [MonoGame/MonoGame#8545] [General]
### Remaining DesktopVK Issues
- Lots of features seem to be unavailable such as mipmaps, instancing, scalar arrays, texture bindings when registers aren't defined in HLSL, and more.
- Lots of "TODO" & "FIXME" comments.
- Lots of "Validation Error" messages in the output window, flickering and at times ~15 second freezes / hardware lockups.
- Lots of unit tests currently fail, are skipped or don't run at all. In total there are 128 cases:

![Screenshot 2024-10-23 194628](https://github.com/user-attachments/assets/1930c46a-a82d-4bb8-a6b7-e7b891f9785f)

### Options for Collating & Tackling Issues
- Add every separate bug as an issue on GitHub (there could be ~100 just with the unit tests though).
- An issue for each category of unit tests group.
- A monolithic DesktopVK issue with a tick list.
- A GitHub discussion.
- A Discord forum post.
- Pair/group/community programming/debugging sessions.
- Free for all.
- A combination of multiple of the above.

Any thoughts?

**Comment by ThomasFOG** [comment]
I don't know if an issue per problem/bug/test should be opened, because there are likely things that, if fixed, would fix multiple of them at once.

Maybe into chunks? E.g. "fixing tests", "fixing validation errors", "missing features"...

@squarebananas @tomspilman thoughts?

Also, should we collate all the issues into one bounty or something like this?

**Comment by squarebananas** [comment]
@mrhelmut Yeah, that's what I was thinking and didn't want to mess up the issues section. Substantial features that are not implemented at all like instancing could probably have their own issue. Firstly though collating a table of issues, whether they're bugs or unimplemented, what has and hasn't been investigated will probably help.

I've figured out a few workarounds for various things (like the flickering stops when only having one frame for the swapchain) but collating any discoveries centrally might help figure out other issues as well.

If a bounty does happen, I'd be happy to team up with whoever and help out anyway. I just don't have as much time as I'd like at the moment but can spare a few hours here and there.

**Comment by tomspilman** [comment]
> D:\Projects\Stardew\stardew_console_mobile_1_6\Brute\Xb1\Durango\Packages
> Substantial features that are not implemented at all like instancing could probably have their own issue. 

Yeah @squarebananas @mrhelmut 

This is the right approach. 

I would stand up the unit tests as soon as we can even if many are failing, just so we can measure our progress.  I think we can set it so that even if the unit tests fail it doesn't fail the PR.  ?

**Reply by ThomasFOG** [reply]
Is it possible to have the tests not failing when the error is ```NotImplementedException```? (or just raise warnings?) That way we can differentiate what's a regression from what's not.
